tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -673,11 +676,11 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
             renderPage am.document, 9
             renderPage am.document, 10
 
-        # Now we trigger the actual action
-        anchor.scrollIntoView().then ->
+        anchor.scrollToView()
+        watchForScroll().then ->
```

Comment:
> If you're calling `scrollToView` before `watchForScroll` then I think `pendingTest` hasn't been initialized yet for this run. 

This is the sequence of events happening:
 * We call the Anchor's `scrollToView()`
 * It triggers a page rendering, but it will only happen a bit later.
 * Then we call `watchForScroll()`, which sets `pendingTest`.
 * When the (fake) page is rendered, the fake Highlight is created, and
 * `scrollToView()` on the Highlight is called,
 * which resolves the promise (using the functions stored in `pendingTest`)
 * and then we run our asserts.
 
> Furthermore, if you're resolving the promise instantaneously in your spy there's no reason to do this in a callback.

The spy is for Highlight's `scrollToView()`. That's instantaneous. However, here we are testing the Anchor's `scrollToView()`, which does *not* complete it's task instantaneously, since it needs to wait for the asynchronous page rendering to happen.

>  I'm not sure what's getting tested here.

The two-phased scrolling behavior, which includes starting scrolling (by triggering a page rendering), then waiting for it to finish, and then finishing the scrolling (by calling `scrollToView`() on the right highlight.).